blocklylib-core/src/main/java/com/google/blockly/android/ui/fieldview/BasicFieldVariableView.java

```diff
@@ -199,10 +200,36 @@ public void run() {
          * @param variableNameManager The name manager containing the variables.
          * @param context A context for inflating layouts.
          * @param resource The {@link TextView} layout to use when inflating items.
+         * @param mSpinner Spinner this is being used in
          */
         public VariableViewAdapter(Context context, NameManager variableNameManager,
-                                   @LayoutRes int resource) {
-            super(context, resource);
+                                   @LayoutRes int resource, AppCompatSpinner mSpinner) {
+            super(context, resource, mSpinner);
+
+            mVariableNameManager = variableNameManager;
+            mVars = mVariableNameManager.getUsedNames();
+
+            mRenameString = context.getString(R.string.rename_variable);
+            mDeleteString = context.getString(R.string.delete_variable);
+            refreshVariables();
+            variableNameManager.registerObserver(new DataSetObserver() {
+                @Override
+                public void onChanged() {
+                    refreshVariables();
+                }
+            });
+        }
+
+        /**
+         * This constructor is for compatibility reasons, or for whe it is not in a Spinner.
+         *
+         * @param variableNameManager The name manager containing the variables.
+         * @param context A context for inflating layouts.
+         * @param resource The {@link TextView} layout to use when inflating items.
+         */
+        public VariableViewAdapter(Context context, NameManager variableNameManager,
+                @LayoutRes int resource) {
+            super(context, resource, null);
```

Comment:
Replace the contents of this constructor with a call to the previous constructor, instead of duplicating code:
```java
this(context, variableNameManager, resource, null);
```

Reply:
Done!